docs/src/imagesegmentation.md

```diff
@@ -0,0 +1,317 @@
+set of seeds in the form of a vector of tuples of `CartesianIndex` and label,
+the algorithm tries to assign these labels to each of the remaining points.
+
+###### Demo
+
+```julia
+julia> using ImageSegmentation, Images;
+julia> img = load("worm.jpg");
+julia> seeds = [(CartesianIndex(104, 48), 1), (CartesianIndex( 49, 40), 1),
+                (CartesianIndex( 72,131), 1), (CartesianIndex(109,217), 1),
+                (CartesianIndex( 28, 87), 2), (CartesianIndex( 64,201), 2),
+                (CartesianIndex(104, 72), 2), (CartesianIndex( 86,138), 2)];
+julia> seg = seeded_region_growing(img, seeds);
+```
+**Original:**
+
+![Original](assets/segmentation/worm.jpg)
+
+**Segmented Image with labels replaced by their intensity means:**
+
+![SegmentedImage](assets/segmentation/worm_seg.jpg)
+
+#### Unseeded Region Growing
+
+This algorithm is similar to [Seeded Region Growing](@ref) but does not require
+any prior information about the seed points. This algorithm checks all the
+boundary points and tries to find the most suitable pixel-label pair. It then
+assigns that pixel to the label and updates the boundary points. If no such
+pair is found, a new label is assigned to that pixel and the list of labels is updated.
+
+###### Demo
+
+```julia
+julia> using ImageSegmentation, Images;
+julia> img = load("scene.jpg");
+julia> seg = unseeded_region_growing(img, 0.08);
+```
+
+| Threshold | Output | Compression percentage|
+| ------------- | ----------| -------------------------|
+| Original    | ![scene](https://user-images.githubusercontent.com/15063205/27087931-ecf2490c-5073-11e7-902f-e28b68975979.jpg) | 0 % |
+| 0.02 | ![seg1](https://user-images.githubusercontent.com/15063205/27088052-41996c74-5074-11e7-9ff1-1941d1be88c4.jpg) | 50% |
+| 0.05 | ![seg2](https://user-images.githubusercontent.com/15063205/27088179-9761458c-5074-11e7-9305-91411d862b22.jpg) | 62.5% |
+| 0.1 | ![seg3](https://user-images.githubusercontent.com/15063205/27088318-f556be88-5074-11e7-9b1e-271963060e90.jpg) | 70.8% |
+| 0.2 | ![seg4](https://user-images.githubusercontent.com/15063205/27088449-4ed7774a-5075-11e7-91bd-f18438ca57a0.jpg) | 79.2% |
+
+
+#### Felzenswalb's Region Merging Algorithm
+
+This algorithm operates on a Region Adjacency Graph (RAG). Each pixel/region is a node in the graph and adjacent pixels/regions have edges between them with weight measuring the dissimilarity between pixels/regions. The algorithm repeatedly merges similar regions till we get the final segmentation. It efficiently computes oversegmented “superpixels” in an image. The function can be directly called with an image (the implementation internally creates a RAG of the image first and then proceeds).
+
+###### Demo
+
+```julia
+using Images, ImageSegmentation, TestImages;
+
+img = Gray.(testimage("house"));
+segments = felzenszwalb(img, 300, 100);
+```
+
+![img1](assets/segmentation/house.jpg) ![img2](assets/segmentation/felzenszwalb.jpg)
+
```

Comment:
Not clear how the segmented image was visualized. It's possible that you should create some visualization functions at the top of this section, e.g., "To visualize each segment in terms of its mean color or intensity, ..." and "To assign a separate color to each segment, ..." and then just call these functions for each visualization.